### PR TITLE
chore: Added deployable build from travis

### PR DESCRIPTION
**Pull request by rjmcneill**
I'll add some comments for things that need changed before merging into react_redux branch.
Leaving these in for now though as it makes it easier to test

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ script:
 
 env:
   global:
-  - GH_REF: github.com/ScottLogic/StockFlux.git
-  - secure: ecJ7bY21fCUBzUnlK11lZWC8lZyXrIxK3jkz6NuI9uEQ0lthmL10s0N+makYA9MJApgm6bgcPQNzJRDZ8ogxMTwMNeD+r4Hj8TcNzI4Mze03KdzGzWv7pY6uW5A2OoonOOLWvIJjK4RZqKTBNbbPt1TWgwS+DY02+QKveUiOsKWKeQ16N1qcH0xPT2cQP98TbioMuBF0hBMekfDmF+RqLPvCzR0Q2QYsh0Yb2uCGdMDQw+VJjlTdNWNaLss+i6DxRjvAm1mecK6OsCsGNJG2fV5DmURSBeP4BD4r+0skLPmYtkiNkJ8bXnDdQwLMbHOwfzxgyE6a2gMdSsR4AQMofjXzFQfwfF/RsC6q7LjvalsytLOJAn1XS7IZ+WRn/PHx7+ceRRCsQYd/dRkWMQgYh6c+cvNSV7yDnnxsbVsrdNP2DIIB+KYLMWD90PH4fH0xlyPvULTjGoJfiKBwjSzNB7Vu8H0M7YtjUD334stI1oTvQwgvN0SA6MWe2BvEBv2avFu5IQ7Kv3mySvlYcFZxHidxTBytKDW809UPs3yHGLCLLvhAnF2BtcvOsqzWKjd5VPUYzGGwEzNeWtiQNyyLNEF5K+/zeeqILEN39YRMQtRjtx8MTzQ9NMz3U3si17htfRHB/R5aO6SWJzABWO+uK3SL2zA6IL5c+9polElssIw=
+  - GH_REF: github.com/rjmcneill/StockFlux.git
```
Comment (rjmcneill) [inline .travis.yml]:
```JavaScript
GH_REF: github.com/ScottLogic/StockFlux.git
```

File: .travis.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ script:
...
+  - secure: HFrxvwK3/0UAZPbawQ3AXOtgN3TLKJnNA5GT2YK13tnEfHFgxQSzXAm+IXOtffnr6GDACNbpn/g8S5latW9oCZfGRmJdBNw2aa+Rpt0Q1eAnXdQEfD6SCo7AtF+WE/KGybV7CBjPNGoMWATpk1hXCQXZy2FD9N8u3pgLWM7Sf9Sxi/TVW4eC49yxdV4s3g9ebgF2Urxaf0LkLNhMFJQdLa56I3SG/dnc+HgmBgWd3fDRKcWjztbG0fpPanMvfh2x7Ri+ooTCwITH/ybZ80MWceKv95mBlV7oysm//RdfvSEpFLntQ4JYS3fzGZbJ2gJxV/yJXShfSuk7K8ueTy4YLVyVgweLfLKUjp1aVSlwyYaxKnF1nRshQm52ZbQBsOwFxAYVuct/F12qHGUdeCnknCiXODoPamF2y/95y8RPtoVGTc5SfbXhLcdbfYdOUORLRvCH6W+i9riezipxPM4+B89x2E9NHX6V+NO98p9tQaT04R3ZhMhJlm6GfYnhLMp0uRspytLpe6REwT97cOuZdcYxl/FabZX09WuB/DYhiV7R1gQR/5WeCTqHE2apEoSCvzkpBk6wqC/RVffRVSWXNytdRTBWzMjY2SkgGc5jSitW5lL0gUwlkE9FABZwlqvUNhr23lhT/j5tfNOKN47JMPoq52GpTzODxHiOR4DeEFI=
```
Comment (rjmcneill) [inline .travis.yml]: `secure: ecJ7bY21fCUBzUnlK11lZWC8lZyXrIxK3jkz6NuI9uEQ0lthmL10s0N+makYA9MJApgm6bgcPQNzJRDZ8ogxMTwMNeD+r4Hj8TcNzI4Mze03KdzGzWv7pY6uW5A2OoonOOLWvIJjK4RZqKTBNbbPt1TWgwS+DY02+QKveUiOsKWKeQ16N1qcH0xPT2cQP98TbioMuBF0hBMekfDmF+RqLPvCzR0Q2QYsh0Yb2uCGdMDQw+VJjlTdNWNaLss+i6DxRjvAm1mecK6OsCsGNJG2fV5DmURSBeP4BD4r+0skLPmYtkiNkJ8bXnDdQwLMbHOwfzxgyE6a2gMdSsR4AQMofjXzFQfwfF/RsC6q7LjvalsytLOJAn1XS7IZ+WRn/PHx7+ceRRCsQYd/dRkWMQgYh6c+cvNSV7yDnnxsbVsrdNP2DIIB+KYLMWD90PH4fH0xlyPvULTjGoJfiKBwjSzNB7Vu8H0M7YtjUD334stI1oTvQwgvN0SA6MWe2BvEBv2avFu5IQ7Kv3mySvlYcFZxHidxTBytKDW809UPs3yHGLCLLvhAnF2BtcvOsqzWKjd5VPUYzGGwEzNeWtiQNyyLNEF5K+/zeeqILEN39YRMQtRjtx8MTzQ9NMz3U3si17htfRHB/R5aO6SWJzABWO+uK3SL2zA6IL5c+9polElssIw=`

File: configJson.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+const configBuilder = require('openfin-config-builder');
+
+if (process.argv.length <= 2) {
+    console.log('Usage: ' + __filename + ' <deployName>');
+    process.exit(-1);
+}
+
+const deployName = process.argv[2];
+const deployedUrl = 'http://rjmcneill.github.io/StockFlux/' + deployName + '/';
```
Comment (rjmcneill) [inline configJson.js]: const deployedUrl = 'http://scottlogic.github.io/StockFlux/' + deployName + '/';

File: deploy.sh [hunk 4]
```diff
@@ -4,30 +4,30 @@ set -eo pipefail
 # Check for release branch - not using grep as set -e means it fails script
 RELEASE_BRANCH=$(echo "$TRAVIS_BRANCH" |  sed -n 's/^release\-/&/p')
 
-#Get the release type (dev/master) from the branch name
+# Get the release type (dev/master) from the branch name
 TYPE="$TRAVIS_BRANCH"
 
-if ([ "$TRAVIS_PULL_REQUEST" == "false" ]  && [ "${TRAVIS_REPO_SLUG}" == "ScottLogic/StockFlux" ] && ([ "$TYPE" == "dev" ] || [ "$TYPE" == "master" ] || [ -n "$RELEASE_BRANCH" ]))
+if ([ "$TRAVIS_PULL_REQUEST" == "false" ]  && [ "${TRAVIS_REPO_SLUG}" == "rjmcneill/StockFlux" ] && ([ "$TYPE" == "chore/deployable-build" ] || [ "$TYPE" == "dev" ] || [ "$TYPE" == "master" ] || [ -n "$RELEASE_BRANCH" ]))
```
Comment (rjmcneill) [inline deploy.sh]: Revert diff

File: deploy.sh [hunk 5]
```diff
@@ -4,30 +4,30 @@ set -eo pipefail
...
 then
-    #Clone the latest gh-pages
-    git clone https://github.com/ScottLogic/StockFlux.git --branch gh-pages gh-pages
+    # Clone the latest gh-pages
+    git clone https://github.com/rjmcneill/StockFlux.git --branch gh-pages gh-pages
```
Comment (rjmcneill) [inline deploy.sh]: Revert diff

File: deploy.sh [hunk 6]
```diff
@@ -4,30 +4,30 @@ set -eo pipefail
...
         echo "Preparing to build version $TYPE"
-        grunt ci --build-target=$TYPE
+        npm run build:deploy $TYPE
 
         rm -rf "./gh-pages/$TYPE"
         cp -r "./public" "./gh-pages/$TYPE"
     fi
 
-    if ([ $TYPE == "master" ] || [ -n "$RELEASE_BRANCH" ])
+    if ([ "$TYPE" == "chore/deployable-build" ] || [ $TYPE == "master" ] || [ -n "$RELEASE_BRANCH" ])
```
Comment (rjmcneill) [inline deploy.sh]: Revert diff

File: deploy.sh [hunk 7]
```diff
@@ -37,27 +37,28 @@ then
...
-    #Removing git history
+    # Removing git history
     rm -rf .git
     git init
 
-    # inside this git repo we'll pretend to be a new user
+    # Inside this git repo we'll pretend to be a new user
     git config user.name "Travis CI"
-    git config user.email "travis@scottlogic.com"
+    git config user.email "travis@rjmcneill.com"
```
Comment (rjmcneill) [inline deploy.sh]: Revert diff

File: configJson.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
+const configBuilder = require('openfin-config-builder');
```
Comment (jleft) [inline configJson.js]: File name is very generic - maybe something like `updateOpenFinConfigForDeployment.js`?

File: configJson.js [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+
+if (process.argv.length <= 2) {
+    console.log('Usage: ' + __filename + ' <deployName>');
```
Comment (jleft) [inline configJson.js]: `console.error` probably more appropriate here

Does `node` need to be used before the file name?

Can the string concatenation be replaced with template strings? (Below too.)

File: configJson.js [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+    process.exit(-1);
```
Comment (jleft) [inline configJson.js]: Typically `1` is used as the exit code for failure - any reason to use `-1` here?

https://nodejs.org/api/process.html#process_process_exit_code

File: configJson.js [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+        applicationIcon: deployedUrl + 'favicon.ico'
+    },
+    shortcut: {
+        icon: deployedUrl + 'favicon.ico'
+    },
+    splashScreenImage: deployedUrl + 'splashscreen.png'
+}, 'public/app.prod.json').then(() => {
+    console.log('Updating of app.prod.json succeeded');
+}).fail((err) => {
+    console.log('Updating of app.prod.json failed:', err);
```
Comment (jleft) [inline configJson.js]: `console.error`

File: configJson.js [hunk 12]
```diff
@@ -0,0 +1,24 @@
+const configBuilder = require('openfin-config-builder');
+
+if (process.argv.length <= 2) {
+    console.log('Usage: ' + __filename + ' <deployName>');
+    process.exit(-1);
+}
+
+const deployName = process.argv[2];
```
Comment (jleft) [inline configJson.js]: Worth validating that this is either a valid semver (see https://www.npmjs.com/package/semver) or `dev` or `master`?

File: deploy.sh [hunk 13]
```diff
@@ -4,30 +4,30 @@ set -eo pipefail
...
 then
-    #Clone the latest gh-pages
-    git clone https://github.com/ScottLogic/StockFlux.git --branch gh-pages gh-pages
+    # Clone the latest gh-pages
+    git clone https://github.com/rjmcneill/StockFlux.git --branch gh-pages gh-pages
 
-    #Get line with version from the file -> get the second word -> remove quotes around the value
-    VERSION=$(grep "version" package.json | awk -v N=$2 '{print $2}' | cut -d \" -f2)
+    # Get line with version from the file -> get the second word -> remove quotes around the value
+    VERSION=$(grep -m 1 "version" package.json | awk -v N=$2 '{print $2}' | cut -d \" -f2)
```
Comment (jleft) [inline deploy.sh]: Would `grep "\"version\":"` work too? Just thinking what might be more robust (i.e. what would happen if the order of the keys in package.json changed).

File: deploy.sh [hunk 14]
```diff
@@ -37,27 +37,28 @@ then
...
         then
-            #For release branches add rc postfix
+            # For release branches add rc postfix
             VERSION="$VERSION-rc"
             echo "Release branch - updating version to $VERSION"
         fi
         # Rebuild everything to do $VERSION
         echo "Cleaning build. Targetting $VERSION"
-        grunt ci --build-target=$VERSION
+        echo "$VERSION"
```
Comment (jleft) [inline deploy.sh]: Remove? Already echoed on the line above

File: package.json [hunk 15]
```diff
@@ -1,11 +1,12 @@
 {
   "name": "bitflux-openfin",
-  "version": "10.1.0",
+  "version": "10.2.0",
```
Comment (jleft) [inline package.json]: Revert

File: updateOpenFinConfigForDeployment.js [hunk 16]
```diff
@@ -0,0 +1,29 @@
+const semver = require('semver');
+const configBuilder = require('openfin-config-builder');
+
+if (process.argv.length <= 2) {
+    console.error(`Usage: node ${__filename} <deployName>`);
+    process.exit(1);
+}
+
+const deployName = process.argv[2];
+const deployedUrl = `http://rjmcneill.github.io/StockFlux/${deployName}/`;
```
Comment (jleft) [inline updateOpenFinConfigForDeployment.js]: I'd move this after the `deployName` validation check and before the OpneFin config update, around line 15 (so it's declared and defined closer to where it's first used).

File: updateOpenFinConfigForDeployment.js [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+
+if ((semver.valid(deployName) !== (deployName)) && (deployName !== 'master') && (deployName !== 'dev') && (deployName !== 'chore/deployable-build')) {
+    process.exit(1);
```
Comment (jleft) [inline updateOpenFinConfigForDeployment.js]: I think it's useful to add an error message here

File: updateOpenFinConfigForDeployment.js [hunk 18]
```diff
@@ -0,0 +1,29 @@
...
+
+if (process.argv.length <= 2) {
+    console.error(`Usage: node ${__filename} <deployName>`);
+    process.exit(1);
+}
+
+const deployName = process.argv[2];
+const deployedUrl = `http://rjmcneill.github.io/StockFlux/${deployName}/`;
+
+if ((semver.valid(deployName) !== (deployName)) && (deployName !== 'master') && (deployName !== 'dev') && (deployName !== 'chore/deployable-build')) {
```
Comment (jleft) [inline updateOpenFinConfigForDeployment.js]: Could possibly make this a bit more descriptive and easier to easier to read? Maybe:

```js
const validDeployableBranchNames = ['master', 'dev', 'chore/deployable-build'];
const isInvalidDeployableBranchName = (name) => validDeployableBranchNames.indexOf(name) === -1;
const isInvalidSemver = (name) => semver.valid(name) === null;

if (isInvalidSemver(deployName) && isInvalidDeployableBranchName(deployName)) {
    // ...
```

**Pull request closed without merging**